core/src/main/scala/za/co/absa/spline/harvester/idGenerators.scala

```diff
@@ -66,15 +63,22 @@ class UUID3IdGenerator[-A <: AnyRef](namespace: UUID) extends HashBasedUUIDGener
 
 class UUID5IdGenerator[-A <: AnyRef](namespace: UUID) extends HashBasedUUIDGenerator[A](namespace, "SHA-1")
 
-class UUID4IdGenerator extends IdGenerator[Any, UUID] {
+class UUID4IdGenerator extends UUIDGenerator[Any] {
   override def nextId(unused: Any): UUID = UUID.randomUUID()
 }
 
-class SequentialIdGenerator(pattern: String) extends IdGenerator[Any, String] {
+class SequentialIdGenerator(pattern: String) {
   private val counter = new AtomicLong(0)
 
-  def nextId(unused: Any): String = {
+  def nextId(): String = {
     val i = counter.getAndIncrement()
     MessageFormat.format(pattern, Long.box(i))
   }
 }
+
+class DataTypeIdGenerator(numberTemplate: String, namespace: UUID) {
+  val seqIdGen = new SequentialIdGenerator(numberTemplate)
+  val UUIDGen = new UUID5IdGenerator[String](namespace)
```

Comment:
```suggestion
  val uuidGen = new UUID5IdGenerator[String](namespace)
```
it's a local variable, so lowercase seems more appropriate

Reply:
otherwise LGTM